src/rendering/luaconsole.cpp

```diff
@@ -886,12 +688,589 @@ void LuaConsole::setCommandInputButton(Key key) {
+            if (jumpCharPos != std::string::npos) {
+                _inputPosition = current.size() - jumpCharPos;
+            }
+            else {
+                _inputPosition = 0;
+            }
+        }
+    );
+
+    // Go to the next JumpCharacter character
+    registerKeyHandler(
+        Key::Right,
+        KeyModifier::Control,
+        [this]() {
+            const std::string current = _commands[_activeCommand];
+            const size_t jumpCharPos = current.find_first_of(
+                JumpCharacters,
+                _inputPosition + 1
+            );
+            if (jumpCharPos != std::string::npos) {
+                _inputPosition = jumpCharPos;
+            }
+            else {
+                _inputPosition = current.size();
+            }
+        }
+    );
+
+    // Go to the previous character
+    registerKeyHandler(
+        Key::Left,
+        KeyModifier::None,
+        [this]() {
+            if (_inputPosition > 0) {
+                --_inputPosition;
+            }
+        }
+    );
+
+    // Go to the previous character
+    registerKeyHandler(
+        Key::B,
+        KeyModifier::Control,
+            [this]() {
+            if (_inputPosition > 0) {
+                --_inputPosition;
+            }
+        }
+    );
+
+    // Go to the next character
+    registerKeyHandler(
+        Key::Right,
+        KeyModifier::None,
+        [this]() {
+            if (!_autoCompleteState.suggestion.empty()) {
+                applySuggestion();
+                return;
+            }
+
+            _inputPosition = std::min(
+                _inputPosition + 1,
+                _commands[_activeCommand].length()
+            );
+        }
+    );
+
+    // Go to the next character
+    registerKeyHandler(
+        Key::F,
+        KeyModifier::Control,
+        [this]() {
+            _inputPosition = std::min(
+                _inputPosition + 1,
+                _commands[_activeCommand].length()
+            );
+        }
+    );
+
+    // Go to previous command
+    registerKeyHandler(
+        Key::Up,
+        KeyModifier::None,
+        [this]() {
+            if (_activeCommand > 0) {
+                --_activeCommand;
+            }
+            _inputPosition = _commands[_activeCommand].length();
+        }
+    );
+
+    // Go to next command (the last is empty)
+    registerKeyHandler(
+        Key::Down,
+        KeyModifier::None,
+        [this]() {
+            if (_activeCommand < _commands.size() - 1) {
+                ++_activeCommand;
+            }
+            _inputPosition = _commands[_activeCommand].length();
+        }
+    );
+
+    // Remove character before _inputPosition
+    registerKeyHandler(
+        Key::BackSpace,
+        KeyModifier::None,
+        [this]() {
+            if (_inputPosition > 0) {
+                _commands[_activeCommand].erase(_inputPosition - 1, 1);
+                --_inputPosition;
```

Comment:
And cases like this as well: 
```suggestion
                _inputPosition--;
```